packages/dds/tree/src/core/tree/pathTree.ts

```diff
@@ -59,6 +60,30 @@ export interface FieldUpPath<TUpPath extends UpPath = UpPath> {
 	readonly field: FieldKey; // TODO: Type information, including when in DetachedField.
 }
 
+/**
+ * An UpPath at the root of a detached tree.
+ */
+export interface DetachedUpPath extends UpPath {
+	/**
+	 * The DetachedNodeId of this path.
+	 */
+	readonly detachedNodeId: DetachedNodeId;
+}
```

Comment:
I wonder if you can add something like:

```ts
readonly parent: UpPath & { parent: undefined };
```

To make it implicit that this path is always underneath a root path?

Reply:
I don't think there's a clean way to do this since the root is a PathNode which in practice has a parent that's undefined but our type definitions allow it to be another PathNode. Since this is low level code, it doesn't really feel worth it to add runtime checks for this, but I don't have strong feelings one way or another

Reply:
Doesn't even matter now cause I removed this interface :)